### PR TITLE
Resolves #153 : add vApp snapshot endpoints.

### LINKED ISSUE #153: Add vApp snapshot endpoint methods

### PR DESCRIPTION
**Pull request by bsnyder788**

### CODE REVIEW

File: src/sdk/model/common/snapshot/has-snapshot.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/* istanbul ignore next: autogenerated */
```
Comment (cfsnyder) [inline src/sdk/model/common/snapshot/has-snapshot.ts]: should not be above imports, should be just above class declaration

File: src/sdk/model/common/snapshot/snapshot-create-request.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/* istanbul ignore next: autogenerated */
```
Comment (cfsnyder) [inline src/sdk/model/common/snapshot/snapshot-create-request.ts]: should not be above imports, should be just above class declaration

File: src/sdk/model/common/snapshot/snapshot-create-request.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import { SnapshotCreateRequestJson } from './__json__/snapshot-create-request-json';
+
+export class SnapshotCreateRequest {
```
Comment (cfsnyder) [inline src/sdk/model/common/snapshot/snapshot-create-request.ts]: add tsdoc string

File: src/sdk/model/common/snapshot/snapshot.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/* istanbul ignore next: autogenerated */
```
Comment (cfsnyder) [inline src/sdk/model/common/snapshot/snapshot.ts]: placed incorrectly

File: src/sdk/model/common/snapshot/snapshot.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import { SnapshotJson } from './__json__/snapshot-json';
+
+export class Snapshot {
```
Comment (cfsnyder) [inline src/sdk/model/common/snapshot/snapshot.ts]: add doc string

File: src/sdk/model/common/snapshot/snapshot.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  constructor(private _json: SnapshotJson) {
+  }
+
+  /**
+   * Get creation date.
+   * @returns {number}
+   */
+  get creationDate(): number {
```
Comment (cfsnyder) [inline src/sdk/model/common/snapshot/snapshot.ts]: you should map that to a JS Date obj

File: src/sdk/model/common/snapshot/has-snapshot.ts (hunk 1 above)
Comment (bsnyder788) [reply src/sdk/model/common/snapshot/has-snapshot.ts]: fixed

File: src/sdk/model/common/snapshot/snapshot-create-request.ts (hunk 2 above)
Comment (bsnyder788) [reply src/sdk/model/common/snapshot/snapshot-create-request.ts]: fixed

File: src/sdk/model/common/snapshot/snapshot.ts (hunk 4 above)
Comment (bsnyder788) [reply src/sdk/model/common/snapshot/snapshot.ts]: fixed

**Pull request merged**